src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1351,9 +1402,67 @@ private void getVars(CORINFO_METHOD_STRUCT_* ftn, ref uint cVars, ILVarInfo** va
             // Just tell the JIT to extend everything.
             extendOthers = true;
         }
+
         private void setVars(CORINFO_METHOD_STRUCT_* ftn, uint cVars, NativeVarInfo* vars)
         {
-            // TODO: Debugging
+            if (_localSlotToInfoMap == null && _parameterIndexToNameMap == null)
+            {
+                return;
+            }
+
+            Dictionary<uint, DebugVarInfo> debugVars = new Dictionary<uint, DebugVarInfo>();
+            int i;
+
+            for (i = 0; i < cVars; i++)
+            {
+                NativeVarInfo nativeVarInfo = vars[i];
+
+                if (nativeVarInfo.varNumber < _parameterIndexToNameMap.Count)
+                {
+                    DebugVarInfo debugVar;
```

Comment:
The code from L#1422 ~ L#1433 seems almost the same as L#1441~L#1454. I suggest to create a function that just is invoked -- you could supply name string to the function depending on param/local.